drivers/start_date_dq/app/views/start_date_dq/warehouse_reports/reports/index.haml

```diff
@@ -0,0 +1,41 @@
+- content_for :title, @report.title
+%h1= @report.title
+= render '/warehouse_reports/breadcrumbs', tab: 'data-quality'
+%p View differences between the client's self-reported date homeless (DateToStreetESSH) and the enrollment entry date.
+= render 'filter'
+- if !@pagy.present?
+  %p.text-center Apply filters to see results.
+- elsif @pagy.count.positive?
+  .d-flex
+    .ml-auto.mb-4
+      = link_to({format: :xlsx}.merge(link_params),{class: 'btn btn-secondary ml-auto'}) do
+        %i.icon-download2
+        Download
+  .d-flex
+    %p!= pagy_info(@pagy, item_name: _('enrollment').pluralize(@pagy.count))
+    %p.ml-auto!= pagy_bootstrap_nav(@pagy)
+  .card
+    %table.mb-0.table.table-striped
+      %thead
+        %tr
+          %th Client
+          - @report.column_names.each do |s|
+            %th= s
+      %tbody
+        - @enrollments.each do |row|
+          - client = row.client
+          - values_hash = @report.column_values(row)
```

Comment:
I don't think you need to set `values_hash` here, you can just reference `@report.column_values(row).each` below.

Reply:
Will do. What's the general rule you're going by here? Only set a variable if it's used more than once/twice/etc in the template?

Reply:
So, no hard and fast rule, but I follow these general guidelines:
Set a variable if
1. it adds clarity to what's happening 
2. it will be used more than once and the method isn't caching the results (so it would have to re-calculate each time)
3. if it's used more than twice (roughly)

Don't set a variable if:
1. it is only used once
2. the method name makes it clear what's going on and doesn't decrease clarity/readability
3. if it is a memory intensive situation and allocating the additional RAM could be an issue

I guess I find setting variables in the view to generally decrease clarity, but I'm not sure why.  In this case, was just noticing that the variable was only used once, so it didn't seem necessary to allocate it.